src/parser/Parser.cpp

```diff
@@ -3,98 +3,105 @@
 Parser::Parser(const Lexer lexer)
     : m_lexer(lexer)
 {
+    m_currenttoken = m_lexer.getToken();
+}
 
+Token Parser::eattoken(TokenType t) {
+    if (m_currenttoken.type != t) {
+        puts("Unexpected token type.");
```

Comment:
`puts` is C-style, can we use iostreams (or even `fprintf`)? 

Also because this is an error I would prefer the output to go to `stderr` instead of `stdout` (e.g. using `cerr`).

Reply:
Yup you're totally right, I was just going quickly here so I could focus on the rest.